sql/core/src/main/scala/org/apache/spark/sql/execution/command/ddl.scala

```diff
@@ -400,6 +400,10 @@ case class AlterTableDropPartition(
       throw new AnalysisException(
         "ALTER TABLE DROP PARTITIONS is not allowed for tables defined using the datasource API")
     }
+    if (specs.size > 1) {
+      throw new AnalysisException(
+        s"Alter Table Drop Partition is not allowed to drop more than one partitions.")
```

Comment:
I see, now we throw an exception here to disallow multiple partitions. So why does this class even take in a sequence of specs int he first place? It should just take in a single spec. We should throw the exception in the parser so the code downstream can be cleaner.